controller/src/main/java/io/pravega/controller/server/Main.java

```diff
@@ -91,6 +91,7 @@ public static void main(String[] args) {
                     .host(Config.REST_SERVER_IP)
                     .port(Config.REST_SERVER_PORT)
                     .tlsEnabled(Config.TLS_ENABLED)
+                    .tlsProtocolVersion(Config.TLS_PROTOCOL_VERSION)
```

Comment:
I don't dispute that validating property values is absolutely a great idea; but isn't it a bigger problem than just this property?  Each of the properties could in theory have a bogus/invalid/incompatible value. Is there a facility already where property values are validated?   

If not, while I fully support the sentiment, I'm concerned about extending the scope of the PR.   Just a thought.

Reply:
We do validate our properties where appropriate. Most of our property values are primitive types (boolean, int, long) so besides number parsing, the only other validation we do is in case some properties must be positive, negative (or greater/smaller than other properties). A new one is also `Duration` where we express time.

We do this in the `*Config` object when we create it. The whole point is that when we pass that `Config` object to the application components, it already contains pre-parsed & validated configuration. There are a few benefits to this, of which the most important ones that come to my mind are:
1. Failing the service early. Do not start the service if config is bad vs detecting the bad config value after the server has started (in which case what do you do? restart? begin throwing and logging errors? ignore?)
2. Keeping parsing and validation logic in one place, which aids with testing and encapsulation. Future changes will mostly require touching one place, not everywhere in the code.

Reply:
Nothing to add here. Makes a lot of sense.

Reply:
I've created enum (TLSProtocolVersion) to parse and validate `tlsProtocolVersion`  variable in both Controller and Segment store sides.